packages/components/src/lib/utility/platform.js

```diff
@@ -32,14 +32,42 @@ export default class Platform {
       this.constructor.evt.ORIENTATION_CHANGE.pub({ orientation: this.orientation })
     }
 
-    this.viewport = {
-      width: window.innerWidth && document.documentElement.clientWidth && document.body.clientWidth
-        ? Math.min(window.innerWidth, document.documentElement.clientWidth, document.body.clientWidth)
-        : document.body.clientWidth || window.innerWidth || document.documentElement.clientWidth,
+    /*
+    If document is in a quirks mode (see https://www.w3.org/TR/2016/WD-cssom-view-1-20160317/#dom-element-clientheight)
+    then `document.body.clientHeight` should be used instead of `document.documentElement.clientHeight`.
+    In a quirks mode a value of `document.compatMode` will be "BackCompat"
+    (see https://developer.mozilla.org/en-US/docs/Web/API/Document/compatMode)
+     */
+    const clientHeight = (document.compatMode === 'BackCompat')
+      ? document.body.clientHeight
+      : document.documentElement.clientHeight
+
+    this.scrollbars = {
+      horizontal: {
+        width: window.innerHeight - clientHeight
+      },
+      vertical: {
+        width: window.innerWidth - document.documentElement.clientWidth
+      }
+    }
 
-      height: window.innerHeight && document.documentElement.clientHeight
-        ? Math.min(window.innerHeight, document.documentElement.clientHeight)
-        : window.innerHeight || document.documentElement.clientHeight
+    this.viewport = {
+      // A width of the viewport that includes the width of the scrollbars, if they are shown
+      width: window.innerWidth,
```

Comment:
is there any chance window.innerWidth is not defined? Is it safe to drop the fallbacks? Same question for innerHeight

Reply:
I did not find any information about cases where it might not behave as expected, except for buggy implementations in some very early versions of some browsers that are now not in use now. I also checked it on several sites known to be problematic, and it worked well there.

I will probably merge it now: it will go through a series of tests anyway, and if there will be any issues, we'll notice and will have time to fix them.

Reply:
sounds good.